pkg/controller/flow/reconcile.go

```diff
@@ -0,0 +1,360 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flow
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/golang/glog"
+	channelsv1alpha1 "github.com/knative/eventing/pkg/apis/channels/v1alpha1"
+	feedsv1alpha1 "github.com/knative/eventing/pkg/apis/feeds/v1alpha1"
+	v1alpha1 "github.com/knative/eventing/pkg/apis/flows/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// TODO: This should come from a configmap
+const defaultBusName = "stub"
+
+// What field do we assume Object Reference exports as a resolvable target
+const targetFieldName = "domainInternal"
+
+const (
+	// SuccessSynced is used as part of the Event 'reason' when a Flow is synced
+	SuccessSynced = "Synced"
+
+	// MessageResourceSynced is the message used for an Event fired when a Flow
+	// is synced successfully
+	MessageResourceSynced = "Flow synced successfully"
+)
+
+var (
+	flowControllerKind = v1alpha1.SchemeGroupVersion.WithKind("Flow")
+)
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Flow resource
+// with the current status of the resource.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	glog.Infof("Reconciling flow %v", request)
+	flow := &v1alpha1.Flow{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, flow)
+
+	if errors.IsNotFound(err) {
+		glog.Errorf("could not find flow %v\n", request)
+		return reconcile.Result{}, nil
+	}
+
+	if err != nil {
+		glog.Errorf("could not fetch Flow %v for %+v\n", err, request)
+		return reconcile.Result{}, err
+	}
+
+	original := flow.DeepCopy()
+
+	flow.Status.InitializeConditions()
+
+	// Reconcile this copy of the Flow and then write back any status
+	// updates regardless of whether the reconcile error out.
+	err = r.reconcile(flow)
+	if equality.Semantic.DeepEqual(original.Status, flow.Status) {
+		// If we didn't change anything then don't call updateStatus.
+		// This is important because the copy we loaded from the informer's
+		// cache may be stale and we don't want to overwrite a prior update
+		// to status with this stale state.
+	} else if _, err := r.updateStatus(flow); err != nil {
+		glog.Warningf("Failed to update flow status: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	// Requeue if the resource is not ready:
+	return reconcile.Result{Requeue: !flow.Status.IsReady()}, err
+}
+
+func (r *reconciler) reconcile(flow *v1alpha1.Flow) error {
+	// See if the flow has been deleted
+	accessor, err := meta.Accessor(flow)
+	if err != nil {
```

Comment:
Why is this necessary? The outer `Reconcile` function already tries to get the flow, so unless something else is calling this, the argument will always be a real object.